scripts/rtma3d/exrtma3d_prepfgs.ksh

```diff
@@ -102,19 +107,49 @@ CDATE=$PDY$cyc
 #    probe_hrrr_guess_nc=$GESINhrrr/conus/hrrr_${hrrrCYCLE}f0${hrrrFHH}
 #    probe_hrrr_guess_nc=$GESINhrrr/hrrr_${hrrrCYCLE}f0${hrrrFHH}
      export probe_hrrr_guess_nc=hrrr_${hrrrCYCLE}f0${hrrrFHH}
+     size_match=no
+     found_hrrrges=no
      if [ -s $GESINhrrr/$probe_hrrr_guess_nc ]; then
          found_hrrrges=yes
-#        cpreq ${probe_hrrr_guess_nc} $COMOUT/${RUN}.t${cyc}z.hrrr_${hrrrCYCLE}f0${hrrrFHH}
-         cpreq $GESINhrrr/${probe_hrrr_guess_nc} ${DATA}/
-         cpreq $GESINhrrr/${probe_hrrr_guess_nc} ${DATA}/${FGSrtma3d_FNAME}
-        break
+
+         ics=1
+#        loop of checking the filesize of hrrr forecast
+         while [ $ics -le ${ics_max} ] ; do
+             filesize=$(stat -c %s $GESINhrrr/$probe_hrrr_guess_nc)
+             if [[ ${filesize} -eq ${targetsize_hrrr} ]] ; then
+                 size_match="yes"
+#                cpreq ${probe_hrrr_guess_nc} $COMOUT/${RUN}.t${cyc}z.hrrr_${hrrrCYCLE}f0${hrrrFHH}
+#                cpreq $GESINhrrr/${probe_hrrr_guess_nc} ${DATA}/${FGSrtma3d_FNAME}
+                 break  # breaking out the loop of checking file size
+             else
+                 size_match="no"
+                 msg="${probe_hrrr_guess_nc} filesize (${filesize}) does not match the standard size (${targetsize_hrrr}). Sleep for 60 seconds and check again ..."
+                 ${ECHO} "${msg}"
+                 sleep ${sleep_time}
+             fi
+             let "ics=ics+1"
+         done
+
+         if [[ ${size_match} =~ [yYtT] ]] ; then
+             cpreq $GESINhrrr/${probe_hrrr_guess_nc} ${DATA}/${FGSrtma3d_FNAME}
+             break      # breaking out the loop of searching for firstguess in HRRR forecast
+         else
+             msg="hrrr ${ic}-hour forecast file ${probe_hrrr_guess_nc} exists, but its filesize (${filesize}) does not match the standard size (${targetsize_hrrr}) even after waiting for ${ics_max} minutes. Trying to search in the earlier hrrr forecast files ..."
```

Comment:
Did you test the option where the filesize did not match the standard size after waiting 15 minutes?  If not, I will test with the NCO compliance work.

Reply:
Hi @AnnetteGibbs-NOAA ,

You got me. I have not tested these lines in real run, because there are very rare case that the file is incomplete. I need to make a fake incomplete file and put it under another directory to test (since I could not put it under hrrr prod directory). 
This is a good question. I do need to test it. I will make a fake one to test it. I will let you know how it goes.

Thanks
-Gang

Reply:
Or I can just test it when I do the compliance work.  Either way works for me.  Thanks Gang!

Reply:
@AnnetteGibbs-NOAA 
I am building up the parallel on WCOSS2 now. If I can make the test with the fake file soon, I will let you know.
Thank you!
-Gang